ros2node/test/test_node.py

```diff
@@ -34,3 +36,28 @@ def test_parse_node_name():
     assert name.full_name == '/ns/talker'
     assert name.namespace == '/ns'
     assert name.name == 'talker'
+
+
+def test_duplicate_names():
+    all_unique_names = [
+        NodeName('foo', 'ns_foo', 'ns_foo/foo'),
+        NodeName('bar', 'ns_foo', 'ns_foo/bar'),
+    ]
+
+    any_duplicates, printable_names = annotate_duplicate_nodes(all_unique_names)
+    assert not any_duplicates
+    assert printable_names == [
+        'ns_foo/bar',
+        'ns_foo/foo',
+    ]
+
+    all_unique_names = [
+        NodeName('foo', 'ns_foo', 'ns_foo/foo'),
+        NodeName('foo', 'ns_foo', 'ns_foo/foo'),
+    ]
```

Comment:
I think this would be more readable as 2 separate tests;
* `test_annotate_duplicate_nodes_names_all_unique`
* `test_annotate_duplicate_nodes_names_with_duplicates`